### DISCUSSION TITLE
Issue with Cloudflare CDN Not Working After Setting Up VPN and Domain

### DISCUSSION THREAD

**Discussion by mehtix1** [vpnhood/VpnHood#640] [Q&A]
Hello,

I'm using a VPN service (VPNhood) with a private server that I set up for myself. It was working perfectly until my previous IP got blocked.

Now, I have:

A new IP address
A new domain
Here’s what I did:

I configured my new domain in the panel, and everything works fine.
However, when I enable Cloudflare CDN for my domain, it stops working.
Additional Details:
The domain works fine without Cloudflare.
Once I turn on Cloudflare CDN (Proxy Mode - Orange Cloud), I can’t access the server.
Questions:
Could this be related to my VPNhood server settings?
Do I need to adjust any DNS, firewall, or Cloudflare settings to make it work?

**Comment by trudyhood** [comment]
I am afraid to tell you that CDN is not supported by VpnHood yet. We have some ideas to enable it, but it is no on our priority table. Assigning a valid domain with a valid certificate may help prevent IP detection.

Duplicate of #316

**Comment by mehtix1** [comment]
oh. okay. is there any other settings i can do to make the ip detection just harder? i live in iran and the isp firewalls are too much powerful :))).if we were able to use cdn with your vpn it could be game-changing even with 70 percent update speed...

**Comment by trudyhood** [comment]
I see your point; however, domain filtering is much more common than IP filtering. Using a CDN requires binding the endpoint to a domain, which means the same issue will arise, as your domain name will likely be blocked. In this case, nothing will change, and instead of changing the IP, you will need to keep changing the domain name.

Additionally, as far as I know, it is against many policies to use their service for purposes unrelated to providing http web services.
